File: java/src/main/java/com/tomoarrow/idv/bff/java/controller/OAuthController.java
package com.tomoarrow.idv.bff.java.controller;

import com.tomoarrow.idv.bff.java.service.TokenService;
import com.tomoarrow.idv.client.generated.ApiException;
import com.tomoarrow.idv.client.generated.model.TokenRes;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class OAuthController {
    private final TokenService tokenService;

    public OAuthController(TokenService tokenService) {
        this.tokenService = tokenService;
    }

    @PostMapping("/v1/oauth2/token")
    TokenRes issueClientCredentialsToken() throws ApiException {
        return tokenService.issueClientCredentialsToken();
    }
}

File: kotlin/src/main/kotlin/com/tomoarrow/idv/bff/controller/OAuthController.kt
package com.tomoarrow.idv.bff.controller

import com.tomoarrow.idv.bff.service.TokenService
import kotlinx.coroutines.runBlocking
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class OAuthController(
    private val tokenService: TokenService
) {

    @PostMapping("/v1/oauth2/token")
    fun issueToken(): TokenService.IssueAccessTokenResult = runBlocking {
        tokenService.issueClientCredentialsToken()
    }
}
